src/agent/onefuzz-supervisor/src/commands.rs

```diff
@@ -0,0 +1,176 @@
+    }
+
+    let mut ssh_path =
+        PathBuf::from(env::var("ProgramData").unwrap_or_else(|_| "c:\\programdata".to_string()));
+    ssh_path.push("ssh");
+
+    let mut dsa_path = ssh_path.clone();
+    let mut key_path = ssh_path;
+
+    dsa_path.push("ssh_host_dsa_key");
+    key_path.push("administrators_authorized_keys");
+
+    {
+        let mut file = fs::OpenOptions::new().append(true).open(&key_path).await?;
+        file.write_all(key_info.key.expose_ref().as_bytes()).await?;
+    }
+
+    if key_info.set_permissions {
+        verbose!("removing Authenicated Users permissions to ssh key");
+        let result = Command::new("icacls.exe")
+            .arg(&key_path)
+            .arg("/remove")
+            .arg("NT AUTHORITY/Authenticated Users")
+            .stdin(Stdio::null())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()?
+            .wait_with_output()
+            .await?;
+        if !result.status.success() {
+            bail!(
+                "set authorized_keys ({}) permissions failed: {:?}",
+                key_path.display(),
+                result
+            );
+        }
+
+        verbose!("removing inheritance");
+        let result = Command::new("icacls.exe")
+            .arg(&key_path)
+            .arg("/inheritance:r")
+            .stdin(Stdio::null())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()?
+            .wait_with_output()
+            .await?;
+        if !result.status.success() {
+            bail!(
+                "set authorized_keys ({}) permissions failed: {:?}",
+                key_path.display(),
+                result
+            );
+        }
+
+        verbose!("copying ACL from ssh_host_data_key");
+        let result = Command::new("powershell.exe")
+            .args(&["-ExecutionPolicy", "Unrestricted", "-Command"])
+            .arg(format!(
+                "Get-Acl \"{}\" | Set-Acl \"{}\"",
+                dsa_path.display(),
+                key_path.display()
+            ))
+            .stdin(Stdio::null())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()?
+            .wait_with_output()
+            .await?;
+        if !result.status.success() {
+            bail!(
+                "set authorized_keys ({}) permissions failed: {:?}",
+                key_path.display(),
+                result
+            );
+        }
+    }
+
+    info!("ssh key written: {}", key_path.display());
+
+    Ok(())
+}
+
+#[cfg(target_os = "linux")]
+pub async fn add_ssh_key(key_info: SshKeyInfo) -> Result<()> {
+    let user = match &key_info.user {
+        Some(user) => get_user_by_name(&user),
+        None => get_user_by_uid(get_current_uid()),
+    }
+    .ok_or_else(|| format_err!("unable to find user"))?;
+    info!("adding sshkey:{:?} to user:{:?}", key_info, user);
+
+    let mut ssh_path = user.home_dir().to_owned();
```

Comment:
To avoid confusion and remove unnecessary mutability, how about:
```rust
let home_dir = user.home_dir();
// check if exists...
let ssh_path = home_dir.join(".ssh");
// create if needed, &c